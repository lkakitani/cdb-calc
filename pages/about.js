import Layout from '../components/Layout';

const Page = () => <p>cdb-calc - Author: <a href='https://github.com/lkakitani'>https://github.com/lkakitani</a></p>;

export default Layout(Page);
