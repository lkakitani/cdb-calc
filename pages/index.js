import Layout from '../components/Layout';
import { Main } from '../components/Main';

export const Page = () => (
  <div>
    <h1>Calculadora CDB</h1>
    <Main />
  </div>
);

export default Layout(Page);