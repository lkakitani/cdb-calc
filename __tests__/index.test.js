import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import App from '../pages/index';
import { Page } from '../pages/index';

describe('With Enzyme', () => {
  it('App shows "Calculadora CDB"', () => {
    const app = shallow(<App />);
    expect(app.text()).toEqual('<Header /><Page />');

    const page = shallow(<Page />);
    expect(page.find('h1').text()).toEqual('Calculadora CDB');
  })
})

describe('With Snapshot Testing', () => {
  it('App shows "Calculadora CDB"', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})