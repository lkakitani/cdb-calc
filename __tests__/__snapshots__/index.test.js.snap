// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Calculadora CDB" 1`] = `
<div
  style={
    Object {
      "border": "1px solid #DDD",
      "margin": 20,
      "padding": 20,
    }
  }
>
  <div>
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      Home
    </a>
    <a
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      About
    </a>
  </div>
  <div>
    <h1>
      Calculadora CDB
    </h1>
    <div>
      <form
        onSubmit={[Function]}
      >
        <label>
          Data inicial do investimento:
          <input
            id="startDate"
            onChange={[Function]}
            type="date"
            value=""
          />
        </label>
        <br />
        <label>
          Taxa do CDB:
          <input
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <br />
        <label>
          Data atual:
          <input
            id="endDate"
            onChange={[Function]}
            type="date"
            value=""
          />
        </label>
        <br />
        <input
          type="submit"
          value="Submit"
        />
      </form>
      <br />
      Result: 
      
    </div>
  </div>
</div>
`;
